core/state_processor.go

```diff
@@ -98,7 +99,10 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 		receipts = append(receipts, receipt)
 		allLogs = append(allLogs, receipt.Logs...)
 	}
-
+	var tracingStateDB = vm.StateDB(statedb)
```

Comment:
Why don't we create a hookedStatedb in the first place, ahead of transaction execution?

Now we create a hooked one for each transaction, it looks wasteful for me, perhaps I miss something?